core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt

```diff
@@ -252,17 +254,21 @@ class EvaluatedIndividual<T>(val fitness: FitnessValue,
      */
     fun updateImpactOfGenes(previous : EvaluatedIndividual<T>, mutated : EvaluatedIndividual<T>, mutatedGenes: MutatedGeneSpecification, targetsInfo: Map<Int, EvaluatedMutation>){
 
-        Lazy.assert{mutatedGenes.mutatedIndividual != null}
-
         Lazy.assert {
-            mutatedGenes.mutatedIndividual != null
-            tracking != null
+            mutatedGenes.mutatedIndividual != null && tracking != null
         }
 
+        if(previous.getSizeOfImpact(false) != mutated.getSizeOfImpact(false)){
+            log.info("impacts should be same before updating")
```

Comment:
shouldn't this be a warn? is this a possible bug in EM? or something that could happen in regular scenario?